### ISSUE TITLE
[feature]: recurring tasks

### ISSUE THREAD

**Issue by hacker-h** [makeplane/plane#818]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The feature of recurring tasks allows a scheduling and automation of task/issue creation and updates.
As a user I would like to be able to create a task (e.g. Update github access token) and during creation select that this task shall be recurring e.g. yearly.
Implemention options are various, so here are three:

1. A recurring task is a meta task, which creates subtasks periodically and lists them in the meta task.
2. A recurring task  selfupdates itself and is reused, e.g. yearly by renaming its title/description "2022 yearly update akamai certs" => "2023 yearly update akamai certs"
3. A recurring task is executed once and after completion a new recurring task is created. The new task will after completion create the next one and so on.

The first option could be nice UX if you have a lot of subtasks or short recurrence periods that you later might need to look at quite frequently.
The second option is nice if you want to keep your tickets as few and clean as possible, especially for throwaway oneshot tasks that once executed are no longer of any interest (fire and forget).
The third option is maybe the easiest to implement. Each task independently provides its functionality and once it created the next ticket its job is done. Cleaning up old tickets is no problem. As long as the latest/most relevant task remains, the recurrence flow is continued.

### Why should this be worked on?

Some tasks require periodical (manual) execution, like the rolling of special certificates or secrets (e.g. github fine grained access token) or any other kind of periodically enforced human interventions.

Recurring tasks could be created/updated event-based (e.g. api hooks) or time-based (e.g. via cron expressions).

**Comment by vihar** [comment]
thanks for the detailed draft, @hacker-h. our team will pick this up soon.

**Comment by weir1** [comment]
This will be great

**Comment by voska** [comment]
Just switched over from Asana and noticed this was missing. We'd really like to be able to have recurring tasks!

@vihar is this scheduled for implementation?

**Comment by rossthemusicteacher** [comment]
My team and I would also really love to have recurring tasks.

**Comment by mredodos** [comment]
+1

**Comment by JerryVDP** [comment]
any updates on this?

**Comment by DanielSincere** [comment]
I like this suggested implementation! I would use this. I feel I need this.

**Comment by TomEbeyer** [comment]
+1 on this feature

**Comment by JustinMeans** [comment]
This would be super helpful!

**Comment by julianfbeck** [comment]
I also would like this feature!

**Comment by apatwary12** [comment]
Would love this! Currently we're using the API to do this for a list of tasks

**Comment by mmakoto** [comment]
@vihar any updates on this? This is something we also need.

**Comment by wm-ek** [comment]
we would switch to pro/business if that would be available

**Comment by HeliosAnavatis** [comment]
+1 to this request - it is really helpful to be able to do this.  I've worked around it for now by including an iCal-style recurrence in the task description, and writing a script to iterate over tasks, 'resetting' the ones it finds, and running this regularly (I haven't, like @apatwary12 , put my script into a cron job because I don't need to - I just run it manually when I remember i need to!)

**Comment by liminal-cloud** [comment]
I know this is just a feature to add, but this would get lots of people to switch over. I need recurring tasks and plane has almost everything I want. That and a widget for mobile with properly executed sorting of tasks, and plane would be an absolute killer.

**Comment by theparthacus** [comment]
[it's finally here](https://docs.plane.so/core-concepts/projects/recurring-work-items), folks, and we hope you enjoy using it as much as we did building it.
